opensearch-operator/controllers/cluster_test.go

```diff
@@ -329,37 +328,19 @@ var _ = Describe("Cluster Reconciler", func() {
 				return len(stsList.Items) == 2
 			})
 		})
-		It("should not update the node pool image version", func() {
-			Consistently(func() bool {
-				if err := k8sClient.Get(context.Background(), client.ObjectKeyFromObject(&OpensearchCluster), &OpensearchCluster); err != nil {
-					return false
-				}
-				return OpensearchCluster.Status.Version == "1.0.0"
-			}, consistently, interval).Should(BeTrue())
-			wg := sync.WaitGroup{}
-			for _, pool := range OpensearchCluster.Spec.NodePools {
-				wg.Add(1)
-				By(fmt.Sprintf("checking %s node pool", pool.Component))
-				go func(pool opsterv1.NodePool) {
-					defer GinkgoRecover()
-					defer wg.Done()
+		It("should update the node pool image version", func() {
 
+			for _, pool := range OpensearchCluster.Spec.NodePools {
+				Eventually(func() bool {
 					sts := &appsv1.StatefulSet{}
+					err := k8sClient.Get(context.Background(), client.ObjectKey{Namespace: OpensearchCluster.Namespace, Name: clusterName + "-" + pool.Component}, sts)
+					if err != nil {
+						return false
+					}
 
-					Consistently(func() bool {
-						if err := k8sClient.Get(
-							context.Background(),
-							client.ObjectKey{
-								Namespace: OpensearchCluster.Namespace,
-								Name:      clusterName + "-" + pool.Component,
-							}, sts); err != nil {
-							return false
-						}
-						return sts.Spec.Template.Spec.Containers[0].Image == "docker.io/opensearchproject/opensearch:1.0.0"
-					}, consistently, interval).Should(BeTrue())
-				}(pool)
+					return sts.Spec.Template.Spec.Containers[0].Image == "docker.io/opensearchproject/opensearch:1.1.0"
+				}).Should(BeTrue())
 			}
-			wg.Wait()
 		})
 	})
 	When("A node pool is upgrading", func() {
```

Comment:
Dont we need also a test the checks if the plugin version has updated ?

Reply:
Good idea. Added a test for it.